corehq/apps/users/permissions.py

```diff
@@ -75,3 +75,13 @@ def has_permission_to_view_report(couch_user, domain, report_to_check):
         get_permission_name(HqPermissions.view_report),
         data=report_to_check
     )
+
+
+def get_user_has_permissions_dict(couch_user, domain: str):
```

Comment:
Ya, the only thing is that it doesn't account for if user is an admin which `has_permission` does ([ref](https://github.com/dimagi/commcare-hq/blob/3a44014879131d0b87432bdde1a3d7fa028ead5b/corehq/apps/users/models.py#L442))

Reply:
I think I get what you're saying, something like this?
```
status = {}
dm = web_user.get_domain_membership(...)
permissions = dm.role.permissions
for p_name in HqPermissions.permission_names():
    status[p_name] = dm.role.permissions[p_name]
```
But there's these admin [checks within "has_permission"](https://github.com/dimagi/commcare-hq/blob/2696ffadba5c14b1adf73cb252b878daf08abeb6/corehq/apps/users/models.py#L639-L642) that grants the user permission even if the assigned role does not have permission. I don't want to copy all that logic into `get_user_has_permissions_dict`.

Taking a step back, the reason I decided to do it this way is to avoid passing a WebUser object as an argument for `parse_web_users` to reduce coupling. Maybe that was the wrong decision. Now i'm leaning towards passing in the WebUser and doing the check owner.has_permission(domain, "edit_view_tableu_config"), what do you think?

Reply:
sounds good, made that change here [0b6983a](https://github.com/dimagi/commcare-hq/pull/34468/commits/0b6983a0b1b142c973632bc81f192c53d4b97aa1)